integration_tests/generics_list_01.py

```diff
@@ -0,0 +1,18 @@
+from ltypes import TypeVar, SupportsPlus, SupportsZero, Divisible
+from ltypes import f64, i32
+
+T = TypeVar('T', bound=SupportsPlus|SupportsZero|Divisible)
```

Comment:
What is the meaning of `SupportsPlus`, `SupportsZero` and `Divisible`?

Reply:
Can we write these using expressions as well? I will make the meaning of my question more clear once I know the answer to the above comment.

Reply:
It says that the template T can do such operations. SupportsPlus means you can do T+T -> T. SupprtsZero means you can do T = 0. Divisible means you can do T / 5.

Reply:
Okay. `T + T -> T` makes sense. Why do we want to specify `T = 0` statement and `T/5`. Why can't we do `T = TypeVar('T', bound=T + T -> T | T/T -> T)`? How many more such specifications would be needed. Will we support `SupportsMul`, `SupportsDivide`, `SupportsStringConcat` or in general, `Supports[Op]` where `[Op]` can be replaced the operation the type `T` will support .

Reply:
Yes, that is the plan to just list all the operations: https://github.com/lcompilers/lpython/issues/984, but that will come later. First we wanted an initial constraints operations done, which this PR delivers. Then we will refine it, figure out the syntax and all the details how to just specify operations (and it seems that yes, you need "T = 0" as well), etc.

This PR just needs to be good enough, no bad choices that would be hard to improve upon. But it does not implement the final design, just like previous merged PRs did not implement the final design either, it just moved us towards the final design.

Reply:
@czgdp1807 It's still the prototype for the restriction, I got it working with some hard-coded restrictions, but I believe that we will change the design later so that restriction can be freely defined by users.

Reply:
@czgdp1807 is this ok to merge?

(We still need to polish the history.)

Reply:
Okay. Makes sense. Feel free to merge.